calibrate.py

```diff
@@ -0,0 +1,438 @@
+            self.colour_values[colour] = self.processor.colour_value
+            data = self.colour_values
+            filename = self.file_radio_buttons[self.file_index]['label'] + ".json"
+            with open(filename, 'w') as f:
+                json.dump(data, f)
+            save_label = sgc.Label(text="saved", pos=(15, 160), col=(255,255,255))
+            save_label.add(101)
+            self.clock = pygame.time.Clock()
+            timer = self.clock.tick(30)
+            sgc.update(timer)
+            time.sleep(1)
+            save_label.remove(fade=False) 
+
+    def update_display(self):
+        for radio in self.file_radio_buttons:
+            if radio['btn'].has_focus():
+                self.display_values()
+                self.logger.info("%s file selected for colour editing" % radio['label'])
+                if self.colour_label:
+                    self.colour_label.remove(fade=False)
+        for radio in self.colour_radio_buttons:
+            if radio['btn'].has_focus():
+                self.processor.colour_value = self.colour_values[radio['label']]
+                label_text = str(self.colour_values[radio['label']])
+                if self.colour_label and self.colour_label.active():
+                    self.colour_label.text = label_text
+                else:
+                    self.colour_label = sgc.Label(text=label_text, pos=(15, 150), col=(255,255,255))
+                    self.colour_label.add(100)
+                self.logger.info("%s value selected for editing" % radio['label'])
+
+    def display_files(self, index=None):
+        file_path = os.path.dirname(os.path.realpath(__file__))
+        all_files = os.listdir(file_path)
+        self.file_radio_buttons = []
+        button_index = 0
+        for filename in os.listdir(file_path):
+            if filename.endswith('.json') and filename <> 'calibration.json': 
+                display_name = filename[:len(filename)-5] #trim filetype off
+                button_x = (button_index % 2) * 120 + 10
+                button_y = ((button_index - 1) % 2 + button_index) * 12
+                radio_button = sgc.Radio(group="file", label=display_name, pos=(button_x, button_y), col = (255,255,255))
+                data = dict(btn=radio_button, label=display_name, index=button_index)
+                self.file_radio_buttons.append(data)
+                radio_button.add(button_index)
+                button_index += 1
+        self.file_radio_buttons[index]['btn']._activate() if index is not None else self.file_radio_buttons[0]['btn']._activate()
+
+    def display_values(self, index=None):
+        for button in self.colour_radio_buttons:
+            button['btn'].remove(button['index'])
+        for button in self.file_radio_buttons:
+            if button['btn'].selected:
+                filename = button['label'] + ".json"
+        file_dir = os.path.dirname(os.path.realpath(__file__))
```

Comment:
you're doing this in a couple of methods, but the filepath of the script isn't going to change during runtime. Determine it during `__init__` and then use that throughout, rather than doing it in each method call that needs it?

Reply:
I'm doing this in several classes, let alone methods. make a global variable or pass it around?